### PR TITLE
Fix some leaks found by ASAN.

### PR DESCRIPTION
**Pull request by karliss**
Before PR when opening and closing Cutter with basic executable:
`SUMMARY: AddressSanitizer: 811129 byte(s) leaked in 16725 allocation(s).`
After changes:
`SUMMARY: AddressSanitizer: 588035 byte(s) leaked in 15023 allocation(s).`

### CODE REVIEW

File: libr/core/cconfig.c [hunk 1]
```diff
@@ -1114,39 +1115,31 @@ static bool cb_cfg_fortunes_type(void *user, void *data) {
 	return true;
 }
 
+static void check_decompiler(const char* name) {
```
Comment (ret2libc) [inline libr/core/cconfig.c]: 👍

File: libr/config/config.c [hunk 2]
```diff
@@ -624,6 +624,7 @@ R_API bool r_config_eval(RConfig *cfg, const char *str, bool many) {
 		return true;
 	}
 	__evalString (cfg, s);
+	free(s);
```
Comment (radare) [inline libr/config/config.c]: missing space before (

File: libr/core/canal.c [hunk 3]
```diff
@@ -4072,12 +4072,16 @@ R_API RCoreAnalStats* r_core_anal_get_stats(RCore *core, ut64 from, ut64 to, ut6
 			break;
 		}
 	}
+	r_list_free(metas);
```
Comment (radare) [inline libr/core/canal.c]: missing space before (

File: libr/core/cconfig.c [hunk 4]
```diff
@@ -1114,39 +1115,31 @@ static bool cb_cfg_fortunes_type(void *user, void *data) {
...
 	RConfigNode *node = (RConfigNode *)data;
 	if (node->value[0] == '?') {
 		r_cons_printf ("pdc\n");
 		// spaguetti
-		char *retdec = r_file_path ("r2retdec");
-		if (retdec && *retdec == '/') {
-			r_cons_printf ("!*r2retdec\n");
-			free (retdec);
-		}
+		check_decompiler("r2retdec");
```
Comment (radare) [inline libr/core/cconfig.c]: missing space before (

File: libr/core/cconfig.c [hunk 5]
```diff
@@ -1114,39 +1115,31 @@ static bool cb_cfg_fortunes_type(void *user, void *data) {
...
-		if (r2jadx && *r2jadx == '/') {
-			r_cons_printf ("!*r2jadx\n");
-			free (r2jadx);
-		}
-		char *r2snow = r_file_path ("r2snow");
-		if (r2snow && *r2snow == '/') {
-			r_cons_printf (".!r2snow\n");
-			free (r2snow);
-		}
+		check_decompiler("r2ghidra");
```
Comment (radare) [inline libr/core/cconfig.c]: missing space before (

File: libr/config/config.c (hunk 2 above)
Comment (karliss) [reply libr/config/config.c]: Fixed.

File: libr/core/canal.c (hunk 3 above)
Comment (karliss) [reply libr/core/canal.c]: Fixed.

File: libr/core/cconfig.c (hunk 4 above)
Comment (karliss) [reply libr/core/cconfig.c]: Fixed.

File: libr/core/cconfig.c (hunk 5 above)
Comment (karliss) [reply libr/core/cconfig.c]: Fixed.

**Comment by karliss** [comment]
@radare Sorry, for not noticing style instructions in CONTRIBUTING.md.

**Comment by karliss** [comment]
`SUMMARY: AddressSanitizer: 395111 byte(s) leaked in 11729 allocation(s).`

**Comment by radare** [comment]
thanks!

**Pull request merged**